File: speedsolutionschallenge.client/src/services/hoseService.js
const apiUrl = "api/hose";

const hoseService = {
    createHose: async (hose) => {
        try {
            const response = await fetch(`${apiUrl}/create`, {
                method: "POST",
                headers: {
                    "Content-Type": "application/json",
                },
                body: JSON.stringify(hose),
            });

            if (!response.ok) {
                throw new Error(`Error al añadir la manguera. Código de estado: ${response.status}`);
            }

            return await response.json();
        } catch (error) {
            throw new Error(`Error al añadir la manguera: ${error.message}`);
        }
    },

    getAllHoses: async () => {
        try {
            const response = await fetch(`${apiUrl}/getAll`);
            return await response.json();
        } catch (error) {
            throw new Error(`Error al obtener las mangueras: ${error.message}`);
        }
    },

    getHoseById: async (hoseId) => {
        try {
            const response = await fetch(`${apiUrl}/getById/${hoseId}`);
            return await response.json();
        } catch (error) {
            throw new Error(`Error al obtener la manguera: ${error.message}`);
        }
    },

    updateHose: async (hoseId, updatedHose) => {
        try {
            const response = await fetch(`${apiUrl}/update/${hoseId}`, {
                method: "PUT",
                headers: {
                    "Content-Type": "application/json",
                },
                body: JSON.stringify(updatedHose),
            });

            if (!response.ok) {
                throw new Error(`Error actualizando la manguera. Código de estado: ${response.status}`);
            }

            return await response.json();
        } catch (error) {
            throw new Error(`Error actualizando la manguera: ${error.message}`);
        }
    },

    deleteHose: async (hoseId) => {
        try {
            await fetch(`${apiUrl}/delete/${hoseId}`, {
                method: "DELETE",
            });

            return true;
        } catch (error) {
            throw new Error(`Error al eliminar la manguera: ${error.message}`);
        }
    },
};

export default hoseService;

File: SpeedSolutionsChallenge.Server/Services/Hose/HoseService.cs
using SpeedSolutionsChallenge.Data.Models;
using SpeedSolutionsChallenge.Data.Repositories.HoseRepository;

namespace SpeedSolutionsChallenge.Server.Services.HoseService
{
    public class HoseService : IHoseService
    {
        private readonly IHoseRepository _hoseRepository;

        public HoseService(IHoseRepository hoseRepository)
        {
            _hoseRepository = hoseRepository;
        }

        public async Task<Hose> CreateHose(Hose hose)
        {
            return await _hoseRepository.CreateHose(hose);
        }

        public async Task<List<Hose>> GetAllHoses()
        {
            return await _hoseRepository.GetAllHoses();
        }

        public async Task<Hose> GetHoseById(int hoseId)
        {
            return await _hoseRepository.GetHoseById(hoseId);
        }

        public async Task<Hose> UpdateHose(int hoseId, Hose updatedHose)
        {
            return await _hoseRepository.UpdateHose(hoseId , updatedHose);
        }

        public async Task<bool> DeleteHose(int hoseId)
        {
            return await _hoseRepository.DeleteHose(hoseId);
        }
    }
}
